### PR TITLE
Support for additional metadata with loginSso calls

### PR DESCRIPTION
**Pull request by breedloj**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description
Adds the ability to provide additional metadata with `loginSso` and `reauthConnectionIfNeeded` calls - currently limited to an ability to pass an overridden `credentialSourceId`. This value will be subsequently used in telemetry metrics emitted in this call path.

## Checklist
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/core/credentials/DefaultToolkitAuthManagerTest.kt [hunk 1]
```diff
@@ -3,66 +3,81 @@
...
 import software.aws.toolkits.jetbrains.core.credentials.sso.bearer.InteractiveBearerTokenProvider
+import software.aws.toolkits.jetbrains.core.region.MockRegionProviderExtension
 import software.aws.toolkits.jetbrains.core.region.MockRegionProviderRule
+import software.aws.toolkits.jetbrains.services.telemetry.NoOpPublisher
+import software.aws.toolkits.jetbrains.services.telemetry.TelemetryService
 import software.aws.toolkits.jetbrains.utils.isInstanceOf
 import software.aws.toolkits.jetbrains.utils.isInstanceOfSatisfying
 import software.aws.toolkits.jetbrains.utils.satisfiesKt
 
+@ExtendWith(ApplicationExtension::class)
```
Comment (rli) [inline plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/core/credentials/DefaultToolkitAuthManagerTest.kt:48]: app extension shouldn't be needed if you're using the project one

File: plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/core/credentials/DefaultToolkitAuthManagerTest.kt [hunk 2]
```diff
@@ -3,66 +3,81 @@
...
-    @JvmField
-    @Rule
+    @ExtendWith(MockRegionProviderExtension::class)
     val regionProvider = MockRegionProviderRule()
 
-    @JvmField
-    @Rule
-    val disposableRule = DisposableRule()
+    @TestDisposable
+    private lateinit var disposable: Disposable
```
Comment (rli) [inline plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/core/credentials/DefaultToolkitAuthManagerTest.kt:62]: not sure if idiomatic, but i guess it works

**Review by rli — APPROVED** [review]

File: plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/core/credentials/DefaultToolkitAuthManagerTest.kt (hunk 2 above)
Comment (breedloj) [reply plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/core/credentials/DefaultToolkitAuthManagerTest.kt]: Updated

**Pull request merged**